Error building with Angular 12

### Reproduction link
[https://github.com/NG-ZORRO/ng-zorro-antd](https://github.com/NG-ZORRO/ng-zorro-antd)

### Steps to reproduce
1. create new project
2. add ng-zorro-antd
3. build project

### What is expected?
Successful compilation

### What is actually happening?
Broken build

| Environment | Info |
|---|---|
| ng-zorro-antd | 11.4.1 |
| Browser | Chrome |

---
Following error occurs

```
Error: node_modules/resize-observer-polyfill/src/index.d.ts:19:18 - error TS2717: Subsequent property declarations must have the same type.  Property 'contentRect' must be of type 'DOMRectReadOnly', but here has type 'DOMRectReadOnly'.

19         readonly contentRect: DOMRectReadOnly;
                    ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:12696:14
    12696     readonly contentRect: DOMRectReadOnly;
                       ~~~~~~~~~~~
    'contentRect' was also declared here.
```<!-- generated by ng-zorro-issue-helper. DO NOT REMOVE -->

--- a/components/cdk/resize-observer/package.json
+++ b/components/cdk/resize-observer/package.json
@@ -2,7 +2,6 @@
   "ngPackage": {
     "lib": {
       "entryFile": "public-api.ts"
-    },
-    "allowedNonPeerDependencies": ["tslib", "resize-observer-polyfill"]
+    }
   }
 }

--- a/components/cdk/resize-observer/public-api.ts
+++ b/components/cdk/resize-observer/public-api.ts
@@ -5,4 +5,4 @@
 
 export { NzResizeObserverModule } from './resize-observer.module';
 export { NzResizeObserverDirective } from './resize-observer.directive';
-export { NzResizeObserver, NzResizeObserverFactory as ɵNzResizeObserverFactory } from './resize-observer.service';
+export { NzResizeObserver, NzResizeObserverFactory } from './resize-observer.service';

--- a/components/cdk/resize-observer/resize-observer.service.ts
+++ b/components/cdk/resize-observer/resize-observer.service.ts
@@ -7,8 +7,6 @@ import { coerceElement } from '@angular/cdk/coercion';
 import { ElementRef, Injectable, OnDestroy } from '@angular/core';
 import { Observable, Observer, Subject } from 'rxjs';
 
-import ResizeObserver from 'resize-observer-polyfill';
-
 /**
  * Factory that creates a new ResizeObserver and allows us to stub it out in unit tests.
  */

--- a/components/ng-package.json
+++ b/components/ng-package.json
@@ -5,5 +5,5 @@
   "lib": {
     "entryFile": "./public_api"
   },
-  "allowedNonPeerDependencies": ["tslib", "angular", "date-fns", "@ant-design/icons-angular", "resize-observer-polyfill"]
+  "allowedNonPeerDependencies": ["tslib", "angular", "date-fns", "@ant-design/icons-angular"]
 }

--- a/components/package.json
+++ b/components/package.json
@@ -29,8 +29,7 @@
   "dependencies": {
     "@angular/cdk": "^12.1.0",
     "@ant-design/icons-angular": "^12.0.3",
-    "date-fns": "^2.10.0",
-    "resize-observer-polyfill": "^1.5.1"
+    "date-fns": "^2.10.0"
   },
   "peerDependencies": {
     "@angular/animations": "^12.1.0",

--- a/scripts/site/_site/doc/app/app.module.ts
+++ b/scripts/site/_site/doc/app/app.module.ts
@@ -22,6 +22,7 @@ import { NzPopoverModule } from 'ng-zorro-antd/popover';
 import { NzSelectModule } from 'ng-zorro-antd/select';
 import { ColorSketchModule } from 'ngx-color/sketch';
 import { HoverPreloadModule, HoverPreloadStrategy } from 'ngx-hover-preload';
+import { NzResizeObserverFactory } from 'ng-zorro-antd/cdk/resize-observer';
 
 import { environment } from '../environments/environment';
 import { DEMOComponent } from './_demo/demo.component';
@@ -33,6 +34,8 @@ import { NzContributorsListModule } from './share/contributors-list/contributors
 import { FixedWidgetsModule } from './share/fixed-widgets/fixed-widgets.module';
 import { NzNavBottomModule } from './share/nav-bottom/nav-bottom.module';
 import { SideComponent } from './side/side.component';
+
+import ResizeObserver from 'resize-observer-polyfill';
 
 const icons: IconDefinition[] = [LeftOutline, RightOutline, EditOutline];
 
@@ -71,7 +74,16 @@ const icons: IconDefinition[] = [LeftOutline, RightOutline, EditOutline];
     ),
     ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production && !environment.preProduction })
   ],
-  providers: [Title, { provide: NZ_CONFIG, useValue: { icon: { nzTwotoneColor: '#1890ff' }, global: { nzDirection: 'ltr' } } }],
+  providers: [
+    Title,
+    { provide: NZ_CONFIG, useValue: { icon: { nzTwotoneColor: '#1890ff' }, global: { nzDirection: 'ltr' } } },
+    { provide: NzResizeObserverFactory, useValue: {
+        create(callback: ResizeObserverCallback): ResizeObserver | null {
+          return typeof ResizeObserver === 'undefined' ? null : new ResizeObserver(callback);
+        }
+      }
+    }
+  ],
   bootstrap: [AppComponent]
 })
 export class AppModule {}

--- a/scripts/site/_site/doc/app/header/join-tip.components.ts
+++ b/scripts/site/_site/doc/app/header/join-tip.components.ts
@@ -13,7 +13,7 @@ import { Router } from '@angular/router';
       [nzShowIcon]="false"
     ></nz-alert>
     <ng-template #messageTemplate>
-      🔥阿里云实时计算部前端工程师火热招聘中，<a (click)="navigateToJoin()">点击查看</a>
+      🔥 阿里云开源大数据平台前端工程师火热招聘中，<a (click)="navigateToJoin()">点击查看</a>
     </ng-template>
   `,
   styles: [

